app/features/settings/components/SettingsDrawer.js

```diff
@@ -0,0 +1,168 @@
+
+type Props = {
+
+    /**
+     * Is the drawer open or not.
+     */
+    isOpen: boolean;
+
+    /**
+     * Name of the user.
+     */
+    _name: string;
+
+    /**
+     * Email of the user.
+     */
+    _email: string;
+
+    /**
+     * Avatar URL.
+     */
+    _avatarURL: string;
+
+    /**
+     * Redux dispatch.
+     */
+    dispatch: Dispatch<*>;
+};
+
+/**
+ * Drawer that open when SettingsAction is clicked.
+ */
+class SettingsDrawer extends Component<Props, *> {
+    /**
+     * Initializes a new {@code SettingsDrawer} instance.
+     *
+     * @inheritdoc
+     */
+    constructor() {
+        super();
+
+        this._onBackButton = this._onBackButton.bind(this);
+        this._onNameChange = this._onNameChange.bind(this);
+        this._onEmailBlur = this._onEmailBlur.bind(this);
+        this._onEmailChange = this._onEmailChange.bind(this);
+    }
+
+    /**
+     * Render function of component.
+     *
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <AkCustomDrawer
+                backIcon = { <ArrowLeft label = 'Back' /> }
+                isOpen = { this.props.isOpen }
+                onBackButton = { this._onBackButton } >
+                <DrawerContainer>
+                    <ProfileContainer>
+                        <AvatarContainer>
+                            <Avatar
+                                size = 'xlarge'
+                                src = { this.props._avatarURL } />
+                        </AvatarContainer>
+                        <FieldTextStateless
+                            label = 'Name'
+                            onChange = { this._onNameChange }
+                            shouldFitContainer = { true }
+                            type = 'text'
+                            value = { this.props._name } />
+                        <FieldTextStateless
+                            label = 'Email'
+                            onBlur = { this._onEmailBlur }
+                            onChange = { this._onEmailChange }
+                            shouldFitContainer = { true }
+                            type = 'text'
+                            value = { this.props._email } />
+                    </ProfileContainer>
+                </DrawerContainer>
+            </AkCustomDrawer>
+        );
+    }
+
+    _onNameChange: (*) => void;
+
+    /**
+     * Keeps name input value and name in (redux) state in sync.
+     *
+     * @param {SyntheticInputEvent<HTMLInputElement>} event - Event by which
+     * this function is called.
+     * @returns {void}
+     */
+    _onNameChange(event: SyntheticInputEvent<HTMLInputElement>) {
+        this.props.dispatch(setName(event.currentTarget.value));
+    }
+
+    _onEmailBlur: (*) => void;
+
+    /**
+     * Updates Avatar URL in (redux) state when email is updated.
+     *
+     * @returns {void}
+     */
+    _onEmailBlur() {
+        const avatarURL = getAvatarURL(this.props._email);
```

Comment:
AFAICT we wouldn't be setting any URL if the user enters no email. The way we solve this on Jitsi Meet is by having some locally generated UUID which is used as the key to this function.